### PR TITLE
Add felmat support for amp-ad

### PR DESCRIPTION
**Pull request by felmat**
We want to add felmat Ad-Network into the Supported Ad-Network for amp-ad.

Corporation name: Lombard inc.

Thank you.

### CODE REVIEW

**Comment by felmat** [comment]
I signed it!

2016-11-08 17:21 GMT+09:00 googlebot <notifications@github.com>:

File: examples/ads.amp.html [hunk 1]
```diff
@@ -557,6 +558,17 @@ <h2 id="ezoic">Ezoic</h2>
     <div fallback></div>
   </amp-ad>
 
+  <h2 id="felmat">felmat</h2>
+  <amp-ad width=300 height=250
+      type="felmat"
+      data-host="felmat.net"
+      data-fmt="banner"
+      data-fmk="U12473_n2cJD"
+      data-fmp="">
```
Comment (zhouyx) [inline examples/ads.amp.html]: I see you don't assign a value to `data-fmp`, but you do include it in your example. Is `data-fmp` a required parameters or is it an optional one?
If it's optional I think you can remove it from here and change your doc and `validateData()` function as well.

File: examples/ads.amp.html [hunk 2]
```diff
@@ -557,6 +558,17 @@ <h2 id="ezoic">Ezoic</h2>
     <div fallback></div>
   </amp-ad>
 
+  <h2 id="felmat">felmat</h2>
```
Comment (zhouyx) [inline examples/ads.amp.html]: I am not able to see a test ad with your example provided. I would highly encourage you to update the example here to provide a render-start ad as it helps with testing your PR this time, and manual test later on.

**Comment by felmat** [comment]
@zhouyx Thank you for your review. 
I changed it to set the value of data-fmp to 0.
Please check it.

File: examples/ads.amp.html (hunk 1 above)
Comment (felmat) [inline examples/ads.amp.html]: Thank you for your review. 
I changed it to set the value of data-fmp to 0.
Please check it.

File: examples/ads.amp.html (hunk 2 above)
Comment (felmat) [inline examples/ads.amp.html]: Thank you for your review. 
I showed a test ad.
Please check it.

**Comment by felmat** [comment]
I signed it!

File: examples/ads.amp.html (hunk 2 above)
Comment (zhouyx) [inline examples/ads.amp.html]: @felmat I still can't see the test ad. Got timeout waiting for `render-start`

**Comment by zhouyx** [comment]
@felmat Could you check your test ad? Still can't see an example ad.

**Comment by felmat** [comment]
@zhouyx 
Thanks for your review.
I implemented your request and changed something.
Please check it.

**Comment by zhouyx** [comment]
LGTM, ping @lannka to take a final look.

File: ads/felmat.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import {writeScript, validateData} from '../3p/3p';
+
+/**
+ * @param {!Window} global
+ * @param {!Object} data
+ */
+export function felmat(global, data) {
+  validateData(data, ['host', 'fmt', 'fmk', 'fmp']);
+  global.fmParam = data;
+  writeScript(global, 'https://t.' + encodeURI(data.host) + '/js/fmamp.js');
```
Comment (lannka) [inline ads/felmat.js]: `encodeURIComponent` is preferred for partial URL encoding.

**Comment by lannka** [comment]
LGTM with one  comment.

**Comment by felmat** [comment]
@lannka 
Thanks for your review.
I changed `encodeURI`.
Please check it.

**Comment by zhouyx** [comment]
@felmat Thanks for the Pull Request. Merging.

**Pull request merged**